src/services/arweave/ArweaveCompositeDataProvider.ts

```diff
@@ -53,4 +57,18 @@ export class ArweaveCompositeDataProvider implements SmartweaveContractSource {
       ),
     );
   }
+
+  async getAntConfirmations(id: ArweaveTransactionId) {
+    try {
+      const confirmations = await this.arweave.api
+        .get(`/tx/${id}/status`)
+        .then((res: any) => {
+          return res.data.number_of_confirmations;
+        });
+      return confirmations;
+    } catch (error) {
+      console.error(error);
+      return 0;
+    }
+  }
```

Comment:
the intent of this clas is to call `_providers` that implement the interface and just apply certain strategies on top (`Promise.any`, `Promise.race`) - it SHOULD NOT be calling arweave or anything explicitly

``` suggestion
  async getContractConfirmations(id: ArweaveTransactionId) {
   return Promise.any(this.providers.map(p => p.getContractConfirmations(id));
```

Reply:
updated